### ISSUE TITLE
semidefinite constraint introduces numpy matrix interface error

### ISSUE THREAD

**Issue by jaehyunp** [cvxgrp/cvxpy#280]
Tested on Ubuntu 16.04, both on Python 2.7 and 3.5.

Minimal example:

```
import cvxpy as cvx
import numpy as np

np.random.seed(1)
n = 2

X = cvx.Variable(n, n)
W = np.random.randn(n, n)

obj = cvx.sum_entries(cvx.mul_elemwise(W, X))
cons = [X == 1, X >> 0]
#cons = [X == 1]
prob = cvx.Problem(cvx.Minimize(obj), cons)

print (prob.solve(solver=cvx.MOSEK))
```

The example works fine without the semidefinite constraint `X >> 0`. Alternatively, I can define `X` as `cvx.Semidef(n)` to make it work.

The error messages seemingly have nothing to do with the semidefinite problems:

```
Traceback (most recent call last):
  File "sdp_test.py", line 15, in <module>
    print (prob.solve(solver=cvx.MOSEK))
  File "/usr/local/lib/python2.7/dist-packages/cvxpy-0.3.9-py2.7.egg/cvxpy/problems/problem.py", line 173, in solve
    return self._solve(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/cvxpy-0.3.9-py2.7.egg/cvxpy/problems/problem.py", line 284, in _solve
    self._update_problem_state(results_dict, sym_data, solver)
  File "/usr/local/lib/python2.7/dist-packages/cvxpy-0.3.9-py2.7.egg/cvxpy/problems/problem.py", line 386, in _update_problem_state
    [LeqConstraint, PSDConstraint])
  File "/usr/local/lib/python2.7/dist-packages/cvxpy-0.3.9-py2.7.egg/cvxpy/problems/problem.py", line 467, in _save_dual_values
    self._save_values(result_vec, active_constraints, constr_offsets)
  File "/usr/local/lib/python2.7/dist-packages/cvxpy-0.3.9-py2.7.egg/cvxpy/problems/problem.py", line 495, in _save_values
    rows, cols)
  File "/usr/local/lib/python2.7/dist-packages/cvxpy-0.3.9-py2.7.egg/cvxpy/interface/base_matrix_interface.py", line 111, in block_add
    block = self._format_block(matrix, block, rows, cols)
  File "/usr/local/lib/python2.7/dist-packages/cvxpy-0.3.9-py2.7.egg/cvxpy/interface/base_matrix_interface.py", line 129, in _format_block
    block = self.reshape(block, (rows, cols))
  File "/usr/local/lib/python2.7/dist-packages/cvxpy-0.3.9-py2.7.egg/cvxpy/interface/numpy_interface/matrix_interface.py", line 68, in reshape
    return np.reshape(matrix, size, order='F')
  File "/usr/lib/python2.7/dist-packages/numpy/core/fromnumeric.py", line 225, in reshape
    return reshape(newshape, order=order)
ValueError: total size of new array must be unchanged
```

**Comment by SteveDiamond** [comment]
Yeah, you can't use the ``>>`` operator to make constraints with MOSEK as a solver. Basically the reasons is that getting the SDP dual variables from MOSEK isn't implemented yet. @enzobusseti any ideas about how to get the duals? You can still use ``Semidef(n)`` variables with MOSEK.

**Comment by hadasba1** [comment]
Hi,
I am trying to solve SDP problem. I "build" symetric matrices and I wand to add constrains that each matrix will be possitive semidefinite. But I do not know what is the right syntax:
dim = len(start)
    cubesNum = len(domains)
    t = Variable(cubesNum,1)
    x = Variable(cubesNum - 1, dim)
    iMat = numpy.identity(dim)
    constraints= [0 <= t]
    for i in range(cubesNum-1):
        z = numpy.array([x[i+1,j]-x[i,j] for j in range(dim)])
        z = z.reshape(1,dim)
        col1 = numpy.concatenate((iMat,z),axis=0)
        for j in range(dim):
            col1[j,j] = t[i]
        ti = numpy.array(t[i]).reshape(1,1)
        precol2 =numpy.hstack([z,ti])
        mat = numpy.concatenate((col1,precol2.T), axis =1)
        constraints.append((mat >> 0))  <- Here I get an error

**Comment by hadasba1** [comment]
I tried to use constrains.SDP and get:
Traceback (most recent call last):
  File "C:/Users/hadas/PycharmProjects/projAlg/SDPup.py", line 101, in <module>
    main(numpy.array([0.2,0,0,0,0.3,0,0]),numpy.array([1,0.7,0,0,0.7,0,0]),[[[1],[0],[5]]])
  File "C:/Users/hadas/PycharmProjects/projAlg/SDPup.py", line 61, in main
    prob = Problem(objective,consts)
  File "C:\Users\hadas\Anaconda2\lib\site-packages\cvxpy\problems\problem.py", line 78, in __init__
    self._size_metrics = SizeMetrics(self)
  File "C:\Users\hadas\Anaconda2\lib\site-packages\cvxpy\problems\problem.py", line 672, in __init__
    for var in problem.variables():
  File "C:\Users\hadas\Anaconda2\lib\site-packages\cvxpy\problems\problem.py", line 145, in variables
    vars_ += constr.variables()
AttributeError: 'SDP' object has no attribute 'variables'

**Comment by GrayThomas** [comment]
I'd suggest that until this is fixed, the cvxpy website should really mention this limitation of the (>>, <<) operators. Maybe alongside the other details specific to the MOSEK solver. There is also a really quick solution, if we assume that adding a new semidefinite variables is essentially the same as the original constraint:
```import cvxpy as cvx
import numpy as np

np.random.seed(1)
n = 2
q = cvx.Variable()
X = cvx.Variable(n, n)
W = np.random.randn(n, n)

obj = cvx.sum_entries(cvx.mul_elemwise(W, X))
cons = [X == 1, X >> 0, -X << 0, X>>-np.eye(2)*0.2, X>>q*np.eye(2)]
#cons = [X == 1]

def inspect(con):
	if type(con) is cvx.constraints.psd_constraint.PSDConstraint:
		newcon = con.args[0]-con.args[1]==cvx.Semidef(max(con.args[0].size[0], con.args[1].size[0]))
		return newcon
	return con

prob = cvx.Problem(cvx.Minimize(obj), [inspect(con) for con in cons])

print (prob.solve(solver=cvx.MOSEK, verbose=False))```

**Comment by SteveDiamond** [comment]
This is fixed in 1.0.
